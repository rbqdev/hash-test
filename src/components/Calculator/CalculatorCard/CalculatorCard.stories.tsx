import { Story, Meta } from "@storybook/react/types-6-0";
import CalculatorCard, { ICalculatorCard } from ".";

export default {
  title: "CalculatorCard",
  component: CalculatorCard
} as Meta;

export const Default: Story<ICalculatorCard> = args => (
  <CalculatorCard {...args} />
);
