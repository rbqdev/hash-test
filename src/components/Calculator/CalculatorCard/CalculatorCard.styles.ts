import styled from "styled-components";
import { lighten } from "polished";
import { _constants as themeConstants } from "@styles/themes";
import media from "styled-media-query";

const { THEME_LIGHT } = themeConstants;

export const Wrapper = styled.div`
  width: 608px;
  height: 389px;
  display: flex;
  border: 1px solid
    ${({ theme: { title, colors } }) =>
    title === THEME_LIGHT ? colors.border : lighten(0.1, colors.secondaryBg)};
  border-radius: 4px;
  background: ${({ theme: { title, colors } }) =>
    title === THEME_LIGHT ? colors.white : colors.secondaryBg};
  position: relative;

  ${media.lessThan("medium")`
    width: 100%;
    height: 100%;
    border-radius: 0;
    border: 0;
    overflow: hidden;
  `}
`;
