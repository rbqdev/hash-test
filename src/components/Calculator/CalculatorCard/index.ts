export { default } from "./CalculatorCard";
export type { ICalculatorCard } from "./CalculatorCard";
