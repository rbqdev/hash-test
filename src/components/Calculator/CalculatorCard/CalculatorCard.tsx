import { CalculatorProvider } from "@contexts/Calculator";
import CalculatorForm from "@components/Calculator/CalculatorForm";
import CalculatorDetailsWrapper from "@components/Calculator/CalculatorDetailsWrapper";
import * as Styled from "./CalculatorCard.styles";
import { FiWifiOff } from "react-icons/fi";
interface Props {
  isOffline?: boolean;
}

export default function CalculatorCard({ isOffline = false }: Props) {
  return (
    <CalculatorProvider>
      <Styled.Wrapper>
        <CalculatorForm />
        <CalculatorDetailsWrapper />

        {isOffline && (
          <Styled.MessageOfflineWrapper>
            <Styled.MessageOfflineText>
              <FiWifiOff />
              Você está sem conexão com a internet. Verifique sua conexão e
              tente novamente mais tarde.
            </Styled.MessageOfflineText>
          </Styled.MessageOfflineWrapper>
        )}
      </Styled.Wrapper>
    </CalculatorProvider>
  );
}
