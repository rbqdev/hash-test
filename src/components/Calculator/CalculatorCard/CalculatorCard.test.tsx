import render from "@testsUtils/renderWithTheme";
import CalculatorCard from ".";

describe("Component: CalculatorCard", () => {
  const regexMessage =
    "Você está sem conexão com a internet. Verifique sua conexão e tente novamente mais tarde.";

  it("should render component correctly", () => {
    const { baseElement } = render(<CalculatorCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should NOT render message if network is online", () => {
    const { queryByText } = render(<CalculatorCard />);

    const text = queryByText(regexMessage);

    expect(text).not.toBeInTheDocument();
  });

  it("should render message if network is offline", () => {
    const { queryByText } = render(<CalculatorCard isOffline />);

    const text = queryByText(regexMessage);

    expect(text).toBeInTheDocument();
  });
});
