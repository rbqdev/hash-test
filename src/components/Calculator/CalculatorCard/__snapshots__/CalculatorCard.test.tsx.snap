// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CalculatorCard should render component correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 22px;
}

.c3:last-child {
  margin-bottom: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  color: #656565;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  line-height: 18px;
  margin-bottom: 6px;
}

.c8 {
  color: #cecece;
  font-size: 11px;
  line-height: 14px;
  font-weight: bold;
  margin-top: 4px;
}

.c6 {
  position: relative;
}

.c7 {
  color: #656565;
  background: #fff;
  border: 1px solid #d1dce3;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 14px;
  line-height: 18px;
  padding: 9px 28px 9px 12px;
  width: 100%;
  outline: none;
}

.c7:hover {
  border: 1px solid #b1c4d0;
}

.c7:focus {
  border: 1px solid #66afe9;
}

.c1 {
  width: 375px;
  padding: 41px 70px 44px 56px;
  border-radius: 4px 0 0 4px;
  background: #fff;
}

.c2 {
  margin-bottom: 22px;
  font-size: 24px;
  white-space: nowrap;
  color: #656565;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
}

.c13 {
  color: #5d9cec;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 46px;
}

.c13 > * {
  white-space: nowrap;
}

.c13 h3 {
  font-weight: normal;
  font-size: 1rem;
  margin: 0;
}

.c13 strong {
  margin-left: 3px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: auto;
}

.c10 {
  background: #f7f9fa;
  padding: 40px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-style: italic;
}

.c11 {
  text-transform: uppercase;
  border-bottom: 1px solid #dfecf8;
  color: #3d75bb;
  font-size: 16px;
  font-weight: bold;
  padding-bottom: 4px;
  display: block;
  margin-bottom: 15px;
}

.c0 {
  width: 608px;
  height: 389px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #d1dce3;
  border-radius: 4px;
  background: #fff;
  position: relative;
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
    height: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 25px;
  }

  .c1 form {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c9 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c10 {
    border-radius: 0 4px 4px 0;
  }
}

@media (max-width:768px) {
  .c10 {
    width: 100%;
    height: auto;
    position: absolute;
    bottom: -100%;
    left: 0;
    visibility: hidden;
    z-index: 2;
    -webkit-transition: all 0.2s;
    transition: all 0.2s;
  }
}

@media (max-width:768px) {
  .c14 {
    width: 100%;
    height: 100%;
    background: #00000099;
    position: fixed;
    top: 0;
    left: 0;
    visibility: hidden;
    opacity: 0;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 100%;
    height: 100%;
    border-radius: 0;
    border: 0;
    overflow: hidden;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <section
        class="c1"
      >
        <h2
          class="c2"
        >
          Simule sua Antecipação
        </h2>
        <form>
          <div
            class="c3"
          >
            <label
              class="c4"
              for="ammount"
            >
              <span
                class="c5"
              >
                Informe o valor da venda *
              </span>
              <div
                class="c6"
              >
                <input
                  aria-invalid="false"
                  aria-label="Informe o valor da venda *"
                  aria-required="true"
                  class="c7"
                  id="ammount"
                  name="ammount"
                  type="text"
                />
              </div>
              
            </label>
          </div>
          <div
            class="c3"
          >
            <label
              class="c4"
              for="installments"
            >
              <span
                class="c5"
              >
                Em quantas parcelas *
              </span>
              <div
                class="c6"
              >
                <input
                  aria-invalid="false"
                  aria-label="Em quantas parcelas *"
                  aria-required="true"
                  class="c7"
                  id="installments"
                  name="installments"
                  type="text"
                />
              </div>
              <small
                class="c8"
              >
                Máximo de 12 parcelas
              </small>
            </label>
          </div>
          <div
            class="c3"
          >
            <label
              class="c4"
              for="mdr"
            >
              <span
                class="c5"
              >
                Informe o percentual de MDR *
              </span>
              <div
                class="c6"
              >
                <input
                  aria-invalid="false"
                  aria-label="Informe o percentual de MDR *"
                  aria-required="true"
                  class="c7"
                  id="mdr"
                  name="mdr"
                  type="text"
                />
              </div>
              
            </label>
          </div>
        </form>
      </section>
      <article
        class="c9"
      >
        <div
          class="c10"
        >
          <h2
            class="c11"
          >
            Você Receberá:
          </h2>
          <ul
            class="c12"
          >
            <li
              class="c13"
            >
              <h3>
                Amanhã
                :
              </h3>
              <strong>
                R$ 0,00
              </strong>
            </li>
            <li
              class="c13"
            >
              <h3>
                Em 15 dias
                :
              </h3>
              <strong>
                R$ 0,00
              </strong>
            </li>
            <li
              class="c13"
            >
              <h3>
                Em 30 dias
                :
              </h3>
              <strong>
                R$ 0,00
              </strong>
            </li>
            <li
              class="c13"
            >
              <h3>
                Em 90 dias
                :
              </h3>
              <strong>
                R$ 0,00
              </strong>
            </li>
          </ul>
        </div>
        <div
          aria-hidden="true"
          class="c14"
        />
      </article>
    </div>
  </div>
</body>
`;
